consumer_test.go

```diff
@@ -152,10 +172,26 @@ func TestConsumerRebalancingMultiplePartitions(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	offsetResponseNewest0 := new(OffsetResponse)
+	offsetResponseNewest0.AddTopicPartition("my_topic", 0, 1234)
+	leader0.Returns(offsetResponseNewest0)
+
+	offsetResponseOldest0 := new(OffsetResponse)
+	offsetResponseOldest0.AddTopicPartition("my_topic", 0, 0)
+	leader0.Returns(offsetResponseOldest0)
+
+	offsetResponseNewest1 := new(OffsetResponse)
+	offsetResponseNewest1.AddTopicPartition("my_topic", 1, 1234)
+	leader1.Returns(offsetResponseNewest1)
+
+	offsetResponseOldest1 := new(OffsetResponse)
+	offsetResponseOldest1.AddTopicPartition("my_topic", 1, 0)
+	leader1.Returns(offsetResponseOldest1)
+
```

Comment:
you can dedupe these by moving them inside the loop

it might even be worth writing a helper function that takes mockbroker/topic/partition/newest/oldest and returns the two responses.